renku/api/datasets.py

```diff
@@ -451,15 +451,23 @@ def get_relative_url(self, url):
             url = str(url / submodule_url.name)
         return url
 
-    def get_newest_commit_for_dataset(self, dataset):
-        """Gets the newest commit for a dataset or its files."""
+    def dataset_commits(self, dataset, max_results=None):
+        """Gets the newest commit for a dataset or its files.
+
+        Commits are returned sorted from newest to oldest.
+        """
+        kwargs = {}
+
+        if max_results:
+            kwargs['max_count'] = max_results
+
         paths = [(Path(dataset.path) / self.METADATA).resolve()]
 
         paths.extend(f.full_path for f in dataset.files)
 
-        commit = list(self.repo.iter_commits(max_count=1, paths=paths))[0]
+        commits = self.repo.iter_commits(paths=paths, **kwargs)
```

Comment:
Originally I wanted to pass kwargs from `dataset_commits` to `iter_commits` to make it more versatile, but decided against it since kwargs = :nauseated_face: 

I removed it now